app/services/filter_application_choices_for_providers.rb

```diff
@@ -52,12 +52,18 @@ def provider_location(application_choices, provider_location)
       application_choices.where('sites.id' => provider_location)
     end
 
+    def course_subject(application_choices, subject_ids)
+      courses = Course.with_subjects(subject_ids)
+      subject_ids&.any? ? application_choices.joins(:course).merge(courses) : application_choices
```

Comment:
Can we return early if there are no subject_ids given and only do the course lookup when necessary?

Reply:
@richardpattinson agrees

Reply:
This constructs the subject filter related part of the query so there are no additional lookups, but I will update for clarity

<img width="1917" alt="Screenshot 2021-05-21 at 00 11 15" src="https://user-images.githubusercontent.com/159200/119059892-1d88c600-b9c9-11eb-954b-b2a8c8be196e.png">